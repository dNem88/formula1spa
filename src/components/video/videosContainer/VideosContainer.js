import React, {Fragment} from 'react'
import {Outlet, context} from 'react-router-dom'
import Spinner from '../../common/spinner/Spinner'
import styles from './videosContainer.module.css'
import VerticalSlider from '../verticalSlider/VerticalSlider'


function VideosContainer(props) {

    return (
        <div className={styles['main-container']}>
            {!props.videos ? <Spinner/> : 
            <Fragment>
                <Outlet context={props}/>
                <VerticalSlider {...props}/>
            </Fragment>}
        </div> 
    )
}


export default VideosContainer