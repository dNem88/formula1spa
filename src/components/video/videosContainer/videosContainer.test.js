import React from 'react'
import {MemoryRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'
import VideosContainer from './VideosContainer'


jest.mock('../verticalSlider/VerticalSlider', () => () => 'VerticalSlider')


describe('Testing VideosContainer', () => {
    
    it('Render Properly', () => {
        const snapshot = renderer.create(
        <MemoryRouter>
            <VideosContainer videos={'test-value'}/>
        </MemoryRouter>
    ).toJSON()
        expect(snapshot).toMatchSnapshot()
        expect(snapshot.type).toBe('div')
        expect(snapshot.props.className).toBe('main-container')
    })
})