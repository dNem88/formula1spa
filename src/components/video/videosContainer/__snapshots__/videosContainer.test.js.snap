// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing VideosContainer Render Properly 1`] = `
<div
  className="main-container"
>
  VerticalSlider
</div>
`;
